### PR TITLE
Hotfix for rabi period estimate

### LINKED ISSUE #656: Inaccurate fitted amplitude in rabi_amplitude_msr
Hi, 
I've noticed that the fit of the amplitude of the pi pulse in `rabi_amplitude_msr` returns a value that is close, but incorrect.
image
As you may see in the picture, the returned value is 0.436, but the argmax of the fitting curve is 0.466.
I've tried it on several qubits and it always gives a slightly smaller value.

[20231205_spinq10q_.zip](https://github.com/qiboteam/qibocal/files/13577196/20231205_spinq10q_.zip)

### PR DESCRIPTION
**Pull request by andrea-pasquale**
Closes #656.
I think there was a problem with some simplification in the correction formula for the rabi period estimate.

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: src/qibocal/protocols/characterization/rabi/utils.py [hunk 1]
```diff
@@ -213,4 +213,4 @@ def period_correction_factor(phase: float):
 
     """
 
-    return 1 - np.modf(phase / np.pi)[0]
+    return np.round(1 + phase / np.pi) - phase / np.pi
```
Comment (AleCandido) [inline src/qibocal/protocols/characterization/rabi/utils.py:216]:
```suggestion
    x = phase / np.pi
    return np.round(1 + x) - x
```
Any better name than `x` is a further improvement.

**Review by AleCandido — APPROVED** [review]

**Comment by AleCandido** [comment]
My bad, the two functions are actually equal only in a certain range, and I see why you want the current one
round-vs-modf

I should have tested better...

Another option, that is exactly what you're doing, would be:
```py
1.5 - ((x + 0.5) % 1)
```
round-vs-modf-vs-mod

But it is really a matter of taste...

**Comment by andrea-pasquale** [comment]
Thanks for checking @AleCandido.
Indeed I saw the same with behavior with `np.modf`. For the time being let's keep `np.round`.

**Comment by AleCandido** [comment]
Btw, is there why you fixed the codomain between -0.5 and 1.5?

**Comment by andrea-pasquale** [comment]
> Btw, is there a reason why you fixed the codomain between 0.5 and 1.5?

Which codomain?

**Comment by AleCandido** [comment]
> Which codomain?

The y-axis in the plot (the codomain of the function you're modifying in this PR)

**Comment by andrea-pasquale** [comment]
> > Which codomain?
> 
> The y-axis in the plot (the codomain of the function you're modifying in this PR).
> 
> Essentially, the amplitude is fixed to be 1 by `np.round()` (and all its friends), and you're deciding to center the range at 0, and to make `f(0) = 1`.

I see. Maybe it is better to go a step behind. The (half-)period corrected taking into account the phase should be the following:

$$ \frac{T^*}{2} = \underbracket{\Big(  k - \frac{\phi}{\pi}\Big) }_{\xi} \frac{T}{2}  $$

and since I now that this correction will be small I want to find $k \in \mathbb{Z}$ which minimizes this correction. Which means that I need to find $k$ s.t. $k - \frac{\phi}{\pi} \approx 1$. Therefore $k = \text{int}( 1 + \phi / \pi)$ and the total correction will be  $\xi = \text{int}( 1 + \phi / \pi) - \phi/\pi$. The problem as far as I understood is how to treat this conversion to `int`.

Given that we want $k$ to be as close as possible to the float this is why I think that it is better to use `np.round` instead of `np.modf`. Here are more plots.
image
image

**Comment by AleCandido** [comment]
Yes, I keep forgetting it is a multiplicative correction, that's why you want the value to be around 1.

**Review by Edoardo-Pedicillo — APPROVED** [review]

**Pull request merged**